Source/WebCore/animation/CSSPropertyAnimation.cpp

```diff
@@ -4076,9 +4076,17 @@ void CSSPropertyAnimation::blendProperty(const CSSPropertyBlendingClient& client
     );
 }
 
-bool CSSPropertyAnimation::isPropertyAnimatable(CSSPropertyID property)
+bool CSSPropertyAnimation::isPropertyAnimatable(AnimatableProperty property)
 {
-    return property == CSSPropertyCustom || CSSPropertyAnimationWrapperMap::singleton().wrapperForProperty(property);
+    return WTF::switchOn(property,
+        [] (CSSPropertyID propertyId) {
+            return propertyId == CSSPropertyCustom || !!CSSPropertyAnimationWrapperMap::singleton().wrapperForProperty(propertyId);
+        },
+        [] (const AtomString&) {
+            // FIXME: this should only be true for property that are registered custom properties.
+            return true;
```

Comment:
Does this result in some unnecessary work?

Reply:
Yes, but not more so than we currently are doing since we would already return `true` for `CSSPropertyCustom`. This will be addressed in a future patch to work towards closing down https://bugs.webkit.org/show_bug.cgi?id=249293.